### PR TITLE
Issue 28080

### PR DESCRIPTION
**Pull request by discountscott**
Fix costing on quote copy for when metric "wholesalepricecosting" is set.

### CODE REVIEW

**Comment by lokeshsh** [comment]
Unable to create database on the pull request branch,please find the error below:

Cannot install file  /home/xtuple/src/xtuple/scripts/output/build_28080.sql
See errors in  /home/xtuple/src/xtuple/scripts/output/build_28080.sql.log

################################################################################
# ERROR: psql stderr OUTPUT LAST 20 LINES OF: /home/xtuple/src/xtuple/scripts/output/build_28080.sql.log
################################################################################

 psql:/home/xtuple/src/xtuple/scripts/output/build_28080.sql:47: NOTICE:  function xt.add_column(text,text,text,text,text) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28080.sql:239: NOTICE:  function xt.show_search_path() does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28080.sql:270: NOTICE:  extension "plv8" already exists, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28080.sql:374: NOTICE:  function xt.js_init() does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28080.sql:375: NOTICE:  function xt.js_init(pg_catalog.bool) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28080.sql:1828: NOTICE:  drop cascades to function _docinfo(integer,text,boolean)
CONTEXT:  SQL statement "DROP TYPE public._docinfo CASCADE"
PL/pgSQL function dropifexists(text,text,text,boolean) line 125 at EXECUTE statement
psql:/home/xtuple/src/xtuple/scripts/output/build_28080.sql:1842: NOTICE:  drop cascades to function _gettargetdocument(integer,integer)
psql:/home/xtuple/src/xtuple/scripts/output/build_28080.sql:2148: NOTICE:  function itemipsprice(pg_catalog.int4,pg_catalog.int4,pg_catalog.int4,pg_catalog.numeric,pg_catalog.int4,pg_catalog.int4,pg_catalog.int4,date,date,pg_catalog.int4,pg_catalog.int4,pg_catalog.int4) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28080.sql:2239: WARNING:  Type seqiss already exists
psql:/home/xtuple/src/xtuple/scripts/output/build_28080.sql:2346: NOTICE:  view "docinfo" does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28080.sql:2347: NOTICE:  function _docinfo(pg_catalog.int4,text) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28080.sql:2494: NOTICE:  function _gettargetdocument(pg_catalog.int4,pg_catalog.int4) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28080.sql:13341: ERROR:  syntax error at or near "CASE"
LINE 59:     CASE WHEN (fetchMetricText('WholesalePriceCosting') = 't...
             ^
 
################################################################################
# END OF psql stderr LOG OUTPUT ################################################
################################################################################


Command failed:  Error: Command failed: 
    at ChildProcess.exithandler (child_process.js:658:15)
    at ChildProcess.emit (events.js:98:17)
    at maybeClose (child_process.js:766:16)
    at Process.ChildProcess._handle.onexit (child_process.js:833:5) { [Error: Command failed: ] killed: false, code: 3, signal: null }
{ [Error: Command failed: ] killed: false, code: 3, signal: null }

File: foundation-database/public/functions/copyquote.sql [hunk 1]
```diff
@@ -56,7 +56,8 @@ BEGIN
   INSERT INTO quitem
   ( quitem_quhead_id, quitem_linenumber, quitem_itemsite_id,
     quitem_scheddate, quitem_promdate, quitem_qtyord,
-    quitem_price, quitem_custprice, quitem_unitcost,
+    quitem_price, quitem_custprice,
+    CASE WHEN (fetchMetricText('WholesalePriceCosting') = 't') THEN (select (item_listcost) from item where item_id=quitem_item_id) ELSE stdCost(itemsite_item_id) END,
```
Comment (gilmoskowitz) [inline foundation-database/public/functions/copyquote.sql]: Two things here:
- The `case` belongs in the `values` section of the insert, not the list of column names
- The metric check can be simplified to `case when fetchMetricBool('WholesalePriceCosting') then ... end,`

File: foundation-database/public/functions/copyquotetocustomer.sql [hunk 2]
```diff
@@ -121,7 +121,8 @@ BEGIN
          COALESCE(pSchedDate,  quitem_scheddate),
          quitem_promdate,
          quitem_qtyord,
-         quitem_price,  quitem_custprice,  stdCost(itemsite_item_id),
+         quitem_price,  quitem_custprice,
+         CASE WHEN (fetchMetricText('WholesalePriceCosting') = 't') THEN (select (item_listcost) from item where item_id=quitem_item_id) ELSE stdCost(itemsite_item_id) END,
```
Comment (gilmoskowitz) [inline foundation-database/public/functions/copyquotetocustomer.sql]: Same comments as for https://github.com/xtuple/xtuple/pull/2730/files#diff-bc109077e60970e48d48e86054b27328R60

File: foundation-database/public/functions/copyquote.sql (hunk 1 above)
Comment (discountscott) [inline foundation-database/public/functions/copyquote.sql]: @gilmoskowitz 
My goodness I'm losing my mind... I can't believe I committed that. I'll recommit in just a minute

**Pull request merged**